dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/slicing/DataFlowSlicing.scala

```diff
@@ -13,30 +15,42 @@ object DataFlowSlicing {
+      case Some(parallelism) if parallelism == 1 => Executors.newSingleThreadExecutor()
+      case Some(parallelism) if parallelism > 1  => Executors.newWorkStealingPool(parallelism)
+      case _                                     => Executors.newWorkStealingPool()
+
     (config.fileFilter match {
-      case Some(fileName) => cpg.file.nameExact(fileName).ast.isCall
+      case Some(fileName) => cpg.file.nameExact(fileName).method.call
       case None           => cpg.call
-    }).iterator.method.withMethodNameFilter.withMethodParameterFilter.withMethodAnnotationFilter.ast.isCall
-      .flatMap { c =>
-        val sinks           = config.sinkPatternFilter.map(filter => c.argument.code(filter).l).getOrElse(c.argument.l)
-        val sliceNodes      = sinks.iterator.repeat(_.ddgIn)(_.maxDepth(config.sliceDepth).emit).dedup.l
-        val sliceNodesIdSet = sliceNodes.id.toSet
-        // Lazily set up the rest if the filters are satisfied
-        lazy val sliceEdges = sliceNodes
-          .flatMap(_.outE)
-          .filter(x => sliceNodesIdSet.contains(x.inNode().id()))
-          .map { e => SliceEdge(e.outNode().id(), e.inNode().id(), e.label()) }
-          .toSet
-        lazy val slice = Option(DataFlowSlice(sliceNodes.map(cfgNodeToSliceNode).toSet, sliceEdges))
+    }).iterator.method.withMethodNameFilter.withMethodParameterFilter.withMethodAnnotationFilter.call
+      .map(c => exec.submit(new TrackDataFlowTask(config, c)))
+      .flatMap(_.get())
+      .reduceOption { (a, b) => DataFlowSlice(a.nodes ++ b.nodes, a.edges ++ b.edges) }
+  }
+
+  private class TrackDataFlowTask(config: DataFlowConfig, c: Call) extends Callable[Option[DataFlowSlice]] {
 
-        // Filtering
-        sliceNodes match {
-          case Nil                                                                     => None
-          case _ if config.mustEndAtExternalMethod && !sinksEndAtExternalMethod(sinks) => None
-          case _                                                                       => slice
-        }
+    override def call(): Option[DataFlowSlice] = {
+      val sinks           = config.sinkPatternFilter.map(filter => c.argument.code(filter).l).getOrElse(c.argument.l)
+      val sliceNodes      = sinks.iterator.repeat(_.ddgIn)(_.maxDepth(config.sliceDepth).emit).dedup.l
+      val sliceNodesIdSet = sliceNodes.id.toSet
+      // Lazily set up the rest if the filters are satisfied
+      lazy val sliceEdges = sliceNodes
+        .flatMap(_.outE)
+        .filter(x => sliceNodesIdSet.contains(x.inNode().id()))
+        .map { e => SliceEdge(e.outNode().id(), e.inNode().id(), e.label()) }
+        .toSet
+      lazy val slice = Option(DataFlowSlice(sliceNodes.map(cfgNodeToSliceNode).toSet, sliceEdges))
+
+      // Filtering
+      sliceNodes match {
+        case Nil                                                                     => None
+        case _ if config.mustEndAtExternalMethod && !sinksEndAtExternalMethod(sinks) => None
```

Comment:
I'm finding this block to be slow and could be removed if we do `.callee.isExternal` at the top level.